amr-wind/equation_systems/icns/source_terms/GravityForcing.cpp

```diff
@@ -56,9 +52,11 @@ void GravityForcing::operator()(
 
     const auto& rho_arr =
         ((*m_rho).state(field_impl::phi_state(fstate)))(lev).const_array(mfi);
-    const auto& rho0_arr = (*m_rho0)(lev).const_array(mfi);
-    const bool ir0 = is_rho0;
-    const bool ipt = is_pptb;
+    const auto& rho0_arr = m_use_reference_density
+                               ? (*m_rho0)(lev).const_array(mfi)
+                               : amrex::Array4<amrex::Real>();
```

Comment:
Nice approach, I should use this in my PR #953.

Reply:
thanks ;)